Gem/Code/Include/NetworkPrefabSpawnerInterface.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Math/Transform.h>
+#include <AzFramework/Spawnable/SpawnableEntitiesInterface.h>
+
+namespace MultiplayerSample
+{
+    using PrefabSpawnCallbackBeforeActivation = AZStd::function<void(
+        AZStd::shared_ptr<AzFramework::EntitySpawnTicket>,
+        AzFramework::SpawnableEntityContainerView)>;
+
+    using PrefabSpawnCallback = AZStd::function<void(
+        AZStd::shared_ptr<AzFramework::EntitySpawnTicket>,
+        AzFramework::SpawnableConstEntityContainerView)>;
+
+    struct PrefabCallbacks
+    {
+        PrefabSpawnCallbackBeforeActivation m_beforeActivateCallback;
+        PrefabSpawnCallback m_onActivateCallback;
+    };
+
+    class NetworkPrefabSpawnerRequests
+    {
+    public:
+        AZ_RTTI(NetworkPrefabSpawnerRequests, "{82e5cfb5-6a1a-4bd1-b48d-cd817474d611}");
+        virtual ~NetworkPrefabSpawnerRequests() = default;
+
+        virtual void SpawnPrefab(const AZ::Transform& worldTm, const char* assetPath, PrefabCallbacks callbacks) = 0;
```

Comment:
Really needs some comments, I don't fully understand the differences between these three API's.

Reply:
Added comments.